prussia_rt/src/thread.rs

```diff
@@ -0,0 +1,79 @@
+//! Contains functions and structs related to EE threading and runtime debugging.
+
+use core::fmt::Debug;
+
+/// Represents information about the EE registers at a moment in execution.
+/// Usually used on exception-handling to save the register state before
+/// jumping to an exception handler.
+#[repr(C)]
+pub(crate) struct ThreadControlBlock {
+    pub at: u64,
```

Comment:
I can't remember, is this from my code? If it was, it's technically incorrect because the GPRs are actually 128-bit wide...

Reply:
No, I started with 64-bit because that was the only way to keep the compiler and linker happy for some reason. I've now been trying to rewrite it with 128-bit fields but I can't quite figure out why it's failing to run properly. I've done all sorts of tricks but it always results in TLB Misses. I'll continue debugging it and try get it done.

Reply:
I don't suppose you have an IM platform I could talk to you on?

Reply:
If you'd like, I'm on Signal and Discord. Alternatively, is there any other platform that you'd prefer?

Reply:
Signal would be perfect. I'm ravenslofty.79